src/python/gudhi/perslay.py

```diff
@@ -0,0 +1,263 @@
+    r  = tf.get_variable("r", shape=[1, 1, 1],                              initializer=r_init, trainable=train_vect)         if not r_const         else tf.get_variable("r", initializer=tf.reshape(r_init, [1, 1, 1]))
+    bc_inp = tf.expand_dims(inp, -1)
+    norms = tf.norm(bc_inp - mu, ord=q, axis=2)
+    return 1/(1 + norms) - 1/(1 + tf.abs(tf.abs(r)-norms)) 
+
+def rational_layer(inp, num_elements, mean_init, variance_init, alpha_init, mean_const, variance_const, alpha_const, train_vect):
+    """ Rational PersLay """
+    dimension_before, num_pts = inp.shape[2].value, inp.shape[1].value
+    mu = tf.get_variable("m", shape=[1, 1, dimension_before, num_elements], initializer=mean_init, trainable=train_vect)      if not mean_const      else tf.get_variable("m", initializer=tf.reshape(mean_init, [1, 1, dimension_before, num_elements]))
+    sg = tf.get_variable("s", shape=[1, 1, dimension_before, num_elements], initializer=variance_init, trainable=train_vect)  if not variance_const  else tf.get_variable("s", initializer=tf.reshape(variance_init, [1, 1, dimension_before, num_elements]))
+    al = tf.get_variable("a", shape=[1, 1, num_elements],                   initializer=alpha_init, trainable=train_vect)     if not alpha_const     else tf.get_variable("a", initializer=tf.reshape(alpha_init, [1, 1, num_elements]))
+    bc_inp = tf.expand_dims(inp, -1)
+    return 1/tf.pow(1+tf.reduce_sum(tf.multiply(tf.abs(bc_inp - mu), tf.abs(sg)), axis=2), al)
+
+def exponential_layer(inp, num_elements, mean_init, variance_init, mean_const, variance_const, train_vect):
+    """ Exponential PersLay """
+    dimension_before, num_pts = inp.shape[2].value, inp.shape[1].value
+    mu = tf.get_variable("m", shape=[1, 1, dimension_before, num_elements], initializer=mean_init, trainable=train_vect)      if not mean_const      else tf.get_variable("m", initializer=tf.reshape(mean_init, [1, 1, dimension_before, num_elements]))
+    sg = tf.get_variable("s", shape=[1, 1, dimension_before, num_elements], initializer=variance_init, trainable=train_vect)  if not variance_const  else tf.get_variable("s", initializer=tf.reshape(variance_init, [1, 1, dimension_before, num_elements]))
+    bc_inp = tf.expand_dims(inp, -1)
+    return tf.exp(tf.reduce_sum(-tf.multiply(tf.square(bc_inp - mu), tf.square(sg)), axis=2))
+
+# Vectorizations implementing persistence landscapes
+def landscape_layer(inp, num_samples, sample_init, sample_const, train_vect):
+    """ Landscape PersLay """
+    sp = tf.get_variable("s", shape=[1, 1, num_samples], initializer=sample_init, trainable=train_vect) if not sample_const else tf.get_variable("s", initializer=tf.reshape(sample_init, [1,1,num_samples]))
+    return np.sqrt(2) * tf.maximum( .5 * (inp[:, :, 1:2] - inp[:, :, 0:1]) - tf.abs(sp - .5 * (inp[:, :, 1:2] + inp[:, :, 0:1])), np.array([0]))
```

Comment:
As I feared, we need to duplicate a lot of code for tensorflow, and most likely we will need yet more copies for other frameworks :disappointed: 
The implementation here looks much shorter than your other one in vector_methods. What drawbacks does it have? Is it just slower because it always evaluates all diagram points at every sample position (bad with many points close to the diagonal)? On the other hand, using the vector syntax makes it win a big constant factor over the python double loop.